### PR TITLE
feat(user): Add email support to forgot password

### LINKED ISSUE #1833: Forgot your password issue
it throwing  Error: **No account with that username has been found** even when i gave correct email id.can u please tell me what is the issue and how can i solve that

### PR DESCRIPTION
**Pull request by simison**
Adds support for recovering your account using email. Previously this would work only with username. Now both are supported.

Breaking change to API (`username` field changed to `userhandle`), but it's really simple to fix if anyone out there is relying on that e.g. with their mobile apps.

Resolves #1833

### CODE REVIEW

File: modules/users/server/controllers/users/users.password.server.controller.js [hunk 1]
```diff
@@ -28,17 +28,23 @@ exports.forgot = function (req, res, next) {
...
     function (token, done) {
-      if (req.body.username) {
+      if (req.body.userhandle) {
+
+        var userhandle = req.body.userhandle.toLowerCase();
+
         User.findOne({
-          username: req.body.username.toLowerCase()
+          $or: [
+            { username: userhandle },
```
Comment (lirantal) [inline modules/users/server/controllers/users/users.password.server.controller.js]: cast both to string, i.e: `String(userhandle)`

**Comment by lirantal** [comment]
thanks @simison!

File: modules/users/server/controllers/users/users.password.server.controller.js (hunk 1 above)
Comment (mleanos) [inline modules/users/server/controllers/users/users.password.server.controller.js]: Should we keep this feature consistent with the Login, and use `usernameOrEmail`?

Comment (mleanos) [inline modules/users/server/controllers/users/users.password.server.controller.js]: I'm not opposed to changing the local passport configuration to use something like `userhandle` though.

**Comment by simison** [comment]
@lirantal @mleanos done!

- Changed `userhandle` to `usernameOrEmail`
- [Cast to string](https://github.com/meanjs/mean/pull/1834/files#diff-c3841fb7af244bab7b73ea4c3d72055cR33) for db

**Comment by mleanos** [comment]
LGTM. thank you 👍

**Comment by lirantal** [comment]
Ready to merge @simison

**Comment by simison** [comment]
cheers! :thumbsup:

**Comment by simison** [comment]
Hmm, weird, master errored at Travis with:

```
[13:52:51] I/update - chromedriver: setting permissions to 0755 for /home/travis/build/meanjs/mean/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.31
[13:54:59] E/downloader - Connection timeout downloading: https://github.com/mozilla/geckodriver/releases/download/v0.18.0/geckodriver-v0.18.0-linux64.tar.gz. Default timeout is 4 minutes.
[13:54:59] I/update - geckodriver: file exists /home/travis/build/meanjs/mean/node_modules/protractor/node_modules/webdriver-manager/selenium/geckodriver-v0.18.0.tar.gz
[13:54:59] I/update - geckodriver: unzipping geckodriver-v0.18.0.tar.gz
(node:7357) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: ENOENT: no such file or directory, rename '/home/travis/build/meanjs/mean/node_modules/protractor/node_modules/webdriver-manager/selenium/geckodriver' -> '/home/travis/build/meanjs/mean/node_modules/protractor/node_modules/webdriver-manager/selenium/geckodriver-v0.18.0'
(node:7357) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[13:54:59] Finished 'webdriver_update' after 2.15 min
[13:54:59] Starting 'protractor'...
[13:54:59] Finished 'protractor' after 1.23 ms
[13:54:59] Finished 'test' after 2.42 min
(node:7388) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
[13:54:59] I/launcher - Running 1 instances of WebDriver
[13:54:59] E/local - Error code: 135
[13:54:59] E/local - Error message: No update-config.json found. Run 'webdriver-manager update' to download binaries.
[13:54:59] E/local - Error: No update-config.json found. Run 'webdriver-manager update' to download binaries.
    at Local.addDefaultBinaryLocs_ (/home/travis/build/meanjs/mean/node_modules/protractor/built/driverProviders/local.js:41:23)
    at Local.setupDriverEnv (/home/travis/build/meanjs/mean/node_modules/protractor/built/driverProviders/local.js:82:14)
    at Local.setupEnv (/home/travis/build/meanjs/mean/node_modules/protractor/built/driverProviders/driverProvider.js:111:34)
    at q.then (/home/travis/build/meanjs/mean/node_modules/protractor/built/runner.js:335:41)
    at _fulfilled (/home/travis/build/meanjs/mean/node_modules/protractor/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/home/travis/build/meanjs/mean/node_modules/protractor/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/home/travis/build/meanjs/mean/node_modules/protractor/node_modules/q/q.js:796:13)
    at /home/travis/build/meanjs/mean/node_modules/protractor/node_modules/q/q.js:857:14
    at runSingle (/home/travis/build/meanjs/mean/node_modules/protractor/node_modules/q/q.js:137:13)
    at flush (/home/travis/build/meanjs/mean/node_modules/protractor/node_modules/q/q.js:125:13)
[13:54:59] E/launcher - Process exited with error code 135
E2E Tests failed:
{ [Error: protractor exited with code 135]
  message: 'protractor exited with code 135',
  showStack: false,
  showProperties: true,
  plugin: 'gulp-protractor',
  __safety: { toString: [Function: bound ] } }
[13:54:59] [nodemon] exiting
npm ERR! Test failed.  See above for more details.
```

I'll restart the build and see again.

**Comment by simison** [comment]
Aight:

image

Gotta see if that happens again.

**Pull request merged**